src/librustc_mir/transform/simplify.rs

```diff
@@ -172,9 +173,8 @@ impl<'a, 'tcx> CfgSimplifier<'a, 'tcx> {
         }
     }
 
-    // Collapse a goto chain starting from `start`
-    fn collapse_goto_chain(&mut self, start: &mut BasicBlock, changed: &mut bool) {
-        let mut terminator = match self.basic_blocks[*start] {
+    fn extract_terminator(&mut self, bb: BasicBlock) -> Option<Terminator<'tcx>> {
```

Comment:
Can we use a more descriptive name for this? `take_terminator_if_simple_goto` or so?